include/oneapi/dpl/pstl/glue_memory_impl.h

```diff
@@ -226,14 +228,23 @@ destroy(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardIterator __
 
     const auto __is_parallel =
         oneapi::dpl::__internal::__is_parallelization_preferred<_ExecutionPolicy, _ForwardIterator>(__exec);
-    const auto __is_vector =
-        oneapi::dpl::__internal::__is_vectorization_preferred<_ExecutionPolicy, _ForwardIterator>(__exec);
+    using _is_vector_type =
+#if _ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDOWS_BROKEN
+        ::std::conditional_t<
+            oneapi::dpl::__internal::__is_host_execution_policy<::std::decay_t<_ExecutionPolicy>>::value,
+            ::std::false_type,
+            decltype(oneapi::dpl::__internal::__is_vectorization_preferred<_ExecutionPolicy, _ForwardIterator>(
+                __exec))>;
+#else
+        decltype(oneapi::dpl::__internal::__is_vectorization_preferred<_ExecutionPolicy, _ForwardIterator>(__exec));
+#endif // _ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDOWS_BROKEN
+    const _is_vector_type __is_vector;
```

Comment:
Whether  do we want to use `constexpr` here?

Reply:
This is not necessary for avoiding compilation of unseq version when configuration is broken. The type is inferred in compile type. 
I decided to leave `const` so this line is aligned with similar occurrences in this file.